File: src/main/java/no/nav/syfo/config/ApplicationConfig.java
package no.nav.syfo.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.scheduling.TaskScheduler;
import org.springframework.scheduling.annotation.EnableScheduling;
import org.springframework.scheduling.concurrent.ConcurrentTaskScheduler;
import org.springframework.transaction.annotation.EnableTransactionManagement;
import org.springframework.web.client.RestTemplate;

@Configuration
@EnableTransactionManagement
@EnableScheduling
public class ApplicationConfig {

    @Bean
    public TaskScheduler taskScheduler() {
        return new ConcurrentTaskScheduler();
    }

    @Bean
    public RestTemplate restTemplate() {
        return new RestTemplate();
    }
}

File: src/main/kotlin/no/nav/syfo/config/ApplicationConfig.kt
package no.nav.syfo.config

import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Primary
import org.springframework.scheduling.annotation.EnableScheduling
import org.springframework.transaction.annotation.EnableTransactionManagement
import org.springframework.web.client.RestTemplate
import org.springframework.web.reactive.function.client.WebClient

@Configuration
@EnableTransactionManagement
@EnableScheduling
class ApplicationConfig {

    @Primary
    @Bean
    fun restTemplate() = RestTemplate()

    @Bean
    @Qualifier("AzureAD")
    fun restTemplateAzureAd() = RestTemplate()

    @Bean
    fun webClient() = WebClient
        .builder()
        .build()
}
